ui/src/app/pages/admin/admin-lock-request.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import {
+  SemiBoldHeader
+} from 'app/components/headers';
+import {DatePicker, TextArea} from 'app/components/inputs';
+import {Button} from 'app/components/buttons';
+import {workspaceAdminApi} from 'app/services/swagger-fetch-clients';
+import {useState} from 'react';
+import {TooltipTrigger} from 'app/components/popups';
+import colors from 'app/styles/colors';
+
+interface Props {
+  workspace: string,
+  onLock: Function,
+  onClose: Function
+}
+
+export const AdminLockRequest = (props: Props) => {
+  const [requestReason, setRequestReason] = useState('');
+  const [requestDate, setRequestDate] = useState(new Date());
+  const [showError, setShowError] = useState(false);
+
+
+  const disableLockWorkspace = !requestReason ||
```

Comment:
might be easier to think about this one if we define the const as `lockWorkspaceEnabled`

Reply:
For me it was easier to do it the other way around :) i can give it a try

Reply:
Done also added a condition to confirm the date is valid or not

Reply:
nice, thanks